docs-chatbot/app/api/generate-answer/route.ts

```diff
@@ -0,0 +1,202 @@
+    "https://api.openai.com/v1/moderations",
+    {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${apiKey}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        input: query,
+      }),
+    }
+  ).then((res) => res.json());
+
+  const [results] = moderationResponse.results;
+  return results;
+}
+
+async function getEmbedding(query: string, apiKey: string) {
+  const embeddingResponse = await fetch(
+    "https://api.openai.com/v1/embeddings",
+    {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${apiKey}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        model: "text-embedding-ada-002",
+        input: query.replaceAll("\n", " "),
+      }),
+    }
+  );
+
+  if (embeddingResponse.status !== 200) {
+    throw new Error(embeddingResponse.statusText);
+  }
+
+  const {
+    data: [{ embedding }],
+  } = await embeddingResponse.json();
+
+  return embedding;
+}
+
+const getSectionsQuery = e.params(
+  {
+    target: e.OpenAIEmbedding,
+    matchThreshold: e.float64,
+    matchCount: e.int16,
+    minContentLength: e.int16,
+  },
+  (params) => {
+    return e.select(e.Section, (section) => {
+      const dist = e.ext.pgvector.cosine_distance(
+        section.embedding,
+        params.target
+      );
+      return {
+        content: true,
+        tokens: true,
+        dist,
+        filter: e.op(
+          e.op(e.len(section.content), ">", params.minContentLength),
+          "and",
+          e.op(dist, "<", params.matchThreshold)
+        ),
+        order_by: {
+          expression: dist,
+          empty: e.EMPTY_LAST,
+        },
+        limit: params.matchCount,
+      };
+    });
+  }
+);
+
+async function getContext(embedding: number[]) {
+  const sections = await getSectionsQuery.run(client, {
+    target: embedding,
+    matchThreshold: 0.3,
+    matchCount: 8,
+    minContentLength: 20,
+  });
+
+  let tokenCount = 0;
+  let context = "";
+
+  for (let i = 0; i < sections.length; i++) {
+    const section = sections[i];
+    const content = section.content;
+    tokenCount += section.tokens;
+
+    if (tokenCount >= 1500) {
+      tokenCount -= section.tokens;
+      break;
+    }
+
+    context += `${content.trim()}\n---\n`;
+  }
+
+  return context;
+}
+
+function createFullPrompt(query: string, context: string) {
+  const systemMessage = `
+        As an enthusiastic EdgeDB expert keen to assist, respond to questions in
+        markdown, referencing the given EdgeDB sections.
+
+        If unable to help based on documentation, respond with:
+        "Sorry, I don't know how to help with that."`;
+
+  return codeBlock`
+        ${oneLineTrim`${systemMessage}`}
+
+        EdgeDB sections: """
+        ${context}
+        """
+
+        Question: """
+        ${query}
+        """`;
+}
+
+async function getOpenAiAnswer(prompt: string, secretKey: string) {
```

Comment:
For consistency with other function signatures:
```suggestion
async function getOpenAiAnswer(prompt: string, apiKey: string) {
```

Reply:
yes yes, I missed this one when I refactored